File: GithubUserJava/app/src/main/java/com/example/myapplication/data/local/room/FavoriteDao.java
package com.example.myapplication.data.local.room;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.example.myapplication.data.local.entity.FavoriteEntity;

import java.util.List;

@Dao
public interface FavoriteDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(FavoriteEntity fav);

    @Update
    void update(FavoriteEntity fav);

    @Delete
    void delete(FavoriteEntity fav);

    @Query("SELECT * FROM favorite")
    LiveData<List<FavoriteEntity>> getAllFavorite();

    @Query("SELECT * FROM favorite WHERE username = :username")
    LiveData<List<FavoriteEntity>> getUserFavoriteByUsername(String username);
}

File: GithubUserKotlin/app/src/main/java/com/example/myapplication/data/local/room/FavoriteDao.kt
package com.example.myapplication.data.local.room

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.myapplication.data.local.entity.FavoriteEntity

@Dao
interface FavoriteDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(fav: FavoriteEntity)

    @Update
    fun update(fav: FavoriteEntity)

    @Delete
    fun delete(fav: FavoriteEntity)

    @Query("SELECT  * from favorite")
    fun getAllFavorite(): LiveData<List<FavoriteEntity>>

    @Query("SELECT  * from favorite WHERE username = :username")
    fun getUserFavoriteByUsername(username: String): LiveData<List<FavoriteEntity>>

}